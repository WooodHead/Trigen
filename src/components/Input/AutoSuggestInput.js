import React from 'react';
import Autosuggest from 'react-autosuggest';
import styled from 'styled-components';

import inputStyles from './inputStyles';


export const Wrapper = styled.div`
  .react-autosuggest__input {
    ${inputStyles}
  }
`;

const AutosuggestInput = props => {
  return (
    <Wrapper>
      <Autosuggest {...props} />
    </Wrapper>
  );
};

export default AutosuggestInput;
