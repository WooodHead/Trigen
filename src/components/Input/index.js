/**
  * A simple styled controlled input element
  *
  * @namespace Input
  */

import React from 'react';
import styled from 'styled-components';


const StyledInput = styled.input`
  display: block;
  margin: 0 0 10px 0;
  padding: 5px 10px;
  border: 1px solid ${props => props.error ? props.theme.warning : props.theme.outlines};
  border-radius: 1px;
  background: ${props => props.theme.noneWhite};
  color: ${props => props.theme.text};
  outline: 0;

  &:placeholder-shown,
  &:-ms-input-placeholder {
    color: ${props => props.theme.outlines};
  }
`;

const Input = props => {
  return (
    <StyledInput {...props} />
  );
};

export default Input;
