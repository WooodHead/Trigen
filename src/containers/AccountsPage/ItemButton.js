import styled from 'styled-components';

import Button from 'components/Button';


const getBGImage = props => {
  switch(props.mode) {
    case 'save':
      return `${props.theme.primary} url("src/img/icons/Icon_Button_Save.png") center/14px 14px`;
    case 'cancel':
      return `${props.theme.secondary} url("src/img/icons/Icon_Button_Abort.png") center/15px 15px`;
    case 'edit':
      return `${props.theme.primary} url("src/img/icons/Icon_Button_Edit.png") center/15px 15px`;
    case 'delete':
      return `${props.theme.secondary} url("src/img/icons/Icon_Button_Delete.png") center/11px 15px`;
    default:
      return `${props.theme.secondary} url("src/im/icons/Icon_Button_Abort.png") center/15px 15px`;
  }
};

export default styled(Button)`
  margin: 0 5px 0 0;
  width: 23px;
  height: 23px;

  border-radius: 50%;
  background: ${props => getBGImage(props)} no-repeat;
`;
