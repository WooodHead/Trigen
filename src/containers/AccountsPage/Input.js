/**
  * A simple restyled Input component
  */

import styled from 'styled-components';

import Input from 'components/Input';

export default styled(Input)`
  display: inline-block;
  margin-right: 10px;
`;
