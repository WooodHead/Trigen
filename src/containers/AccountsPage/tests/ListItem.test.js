import React from 'react';
import { shallow } from 'enzyme';

import ConnectedListItem, {
  ListItem,
  RestyledLi,
  Span,
} from '../ListItem';
import Input from '../Input';
import ItemButton from '../ItemButton';


const baseProps = {
  account: {
    id: 0,
    fakeID: 'fakeID',
    username: 'ttester',
    domain: 'example.com',
    edit: false,
  },
  accountStatus: {
    isLoading: false,
  },
  fakeID: 'fakeID',
  cancelEdit: jest.fn(),
  deleteAccount: jest.fn(),
  editAccount: jest.fn(),
  handleAccountChange: jest.fn(),
  saveNewAccount: jest.fn(),
  updateAccount: jest.fn(),
};

function setUp(newProps) {
  const props = {
    ...baseProps,
    ...newProps,
  };
  const enzymeWrapper = shallow(<ListItem {...props} />);

  return {
    props,
    enzymeWrapper,
  };
}

function editSetup(newProps) {
  const { props } = setUp(newProps);

  const editProps = {
    ...props,
    account: {
      ...props.account,
      edit: true,
    },
    EditedAccount: { ...props.account, edit: true }
  };

  const enzymeWrapper = shallow(<ListItem {...editProps} />);

  return {
    props: editProps,
    enzymeWrapper
  };
}

describe('<ListItem />', () => {
  it('should render <RestyledLi />', () => {
    const { enzymeWrapper } = setUp();

    expect(enzymeWrapper.find(RestyledLi).length).toBe(1);
  });

  it('should render <Span /> in normal mode', () => {
    const { enzymeWrapper } = setUp();

    expect(enzymeWrapper.find(Span).length).toBe(2);
  });

  it('should render <Input /> components in edit mode', () => {
    const { enzymeWrapper } = editSetup();
    expect(enzymeWrapper.find(Input).length).toMatchSnapshot();
  });

  it('should call editItem', () => {
    const { enzymeWrapper, props } = setUp();

    enzymeWrapper.instance().editItem();
    expect(props.editAccount.mock.calls.length).toBe(1);
  });

  it('should call saveItem and update existing account', () => {
    const { enzymeWrapper, props } = editSetup();

    enzymeWrapper.instance().saveItem();
    expect(props.updateAccount.mock.calls.length).toBe(1);
  });

  it('should call saveItem and save new account', () => {
    const { enzymeWrapper, props } = editSetup({
      account: {
        id: 0,
        fakeID: 'fakeID',
        username: 'ttester',
        domain: 'example.com',
        edit: false,
        new: true,
      },
    });

    enzymeWrapper.instance().saveItem();
    expect(props.saveNewAccount.mock.calls.length).toBe(1);
  });

  it('should call cancelEdit', () => {
    const { enzymeWrapper, props } = editSetup();

    enzymeWrapper.instance().cancelEdit();
    expect(props.cancelEdit.mock.calls.length).toBe(1);
  });

  it('should call handleChange on input change', () => {
    const { enzymeWrapper, props } = editSetup();
    const DomainInput = enzymeWrapper.find(Input).findWhere(n => n.props().name === 'domain').first();

    DomainInput.simulate('change', { target: {
      value: 'test',
      name: 'domain'
    }});

    expect(props.handleAccountChange.mock.calls.length).toBe(1);
    expect(props.handleAccountChange).toHaveBeenCalledWith('fakeID', {
      value: 'test',
      name: 'domain'
    });
  })
});
