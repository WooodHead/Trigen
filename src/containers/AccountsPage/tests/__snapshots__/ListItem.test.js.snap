exports[`<ListItem /> should render <Input /> components in edit mode 1`] = `2`;
